src/sagemaker/training_compiler/config.py

```diff
@@ -132,31 +132,34 @@ def validate(
             ValueError: Raised if the requested configuration is not compatible
                         with SageMaker Training Compiler.
         """
-
-        if "local" not in estimator.instance_type:
-            requested_instance_class = estimator.instance_type.split(".")[
-                1
-            ]  # Expecting ml.class.size
-            if not any(
-                [
-                    requested_instance_class.startswith(i)
-                    for i in cls.SUPPORTED_INSTANCE_CLASS_PREFIXES
-                ]
-            ):
+        if estimator.instance_type:
+            if "local" not in estimator.instance_type:
+                requested_instance_class = estimator.instance_type.split(".")[
+                    1
+                ]  # Expecting ml.class.size
+                if not any(
+                    [
+                        requested_instance_class.startswith(i)
+                        for i in cls.SUPPORTED_INSTANCE_CLASS_PREFIXES
+                    ]
+                ):
+                    error_helper_string = (
```

Comment:
can we add a warning/info logging to inform customers that `p3dn.24xlarge` & `p4d.24xlarge` instances offer the best performance for distributed training?

Reply:
Do we just want to generalize to EFA enabled instances or what is special about these 2 instance types ?

Reply:
we want to start with recommending EFA instances we have benchmarked, we can add other instances to the list once we benchmark them.